augur/augur_client.py

```diff
@@ -314,7 +317,7 @@ def _contracts_from_abi(self, abi):
     # blockchain until the latest Ethereum block determined at the time of
     # execution. It will then listen for new blocks until the desired end_block
     # has been reached.
-    filter_params = filter
+    filter_params = dict(filter)
```

Comment:
Making the copy here is too late, the caller might have already modified the filter before execution reaches line 320.

Reply:
I have updated this so that the filter is copied before an asyncio task is created. This way the filter is copied after the parameters have been validated and before running `_filter_blocks`.